src/Symfony/Component/VarExporter/Hydrator.php

```diff
@@ -61,15 +61,14 @@ public static function hydrate(object $instance, array $properties = [], array $
             $propertyScopes = InternalHydrator::$propertyScopes[$class] ??= InternalHydrator::getPropertyScopes($class);
 
             foreach ($properties as $name => &$value) {
-                [$scope, $name] = $propertyScopes[$name] ?? [$class, $name];
-                $scopedProperties[$scope][$name] = &$value;
+                [$scope, $name, $readonlyScope] = $propertyScopes[$name] ?? [$class, $name, $class];
+                $scopedProperties[$readonlyScope ?? $scope][$name] = &$value;
             }
-            unset($value);
         }
 
-        foreach ($scopedProperties as $class => $properties) {
+        foreach ($scopedProperties as $scope => $properties) {
```

Comment:
Shouldn't `$properties` be renamed here to something different? Currently, they clash with parameters in the method.

Reply:
At this step $properties (the argument) has been processed and should not be used anymore. I prefer reusing the name since it's the best fit for this loop.